pkg/controller/job/job_controller.go

```diff
@@ -1070,8 +1098,9 @@ func (jm *Controller) trackJobStatusAndRemoveFinalizers(ctx context.Context, job
 			finishedCond = newFailedConditionForFailureTarget(finishedCond, jm.clock.Now())
 		}
 	}
+	syncJobContext.podFailureCountByPolicyAction = podFailureCountByPolicyAction
```

Comment:
it doesn't look like this is used anywhere else, so maybe just pass it as an argument.

Reply:
Yeah, I wasn't sure about it. Passed as an argument.